src/redemptions/redemption.controller.ts

```diff
@@ -0,0 +1,64 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import {
+  Body,
+  Controller,
+  ForbiddenException,
+  Get,
+  HttpStatus,
+  NotFoundException,
+  Post,
+  UseGuards,
+  ValidationPipe,
+} from '@nestjs/common';
+import { ApiExcludeEndpoint, ApiTags } from '@nestjs/swagger';
+import { MagicLinkGuard } from '../auth/guards/magic-link.guard';
+import { Context } from '../common/decorators/context';
+import { MagicLinkContext } from '../common/interfaces/magic-link-context';
+import { RedemptionDto } from './dto/redemption.dto';
+import { SerializedRedemption } from './interfaces/serializedRedemption';
+import { RedemptionService } from './redemption.service';
+import { serializeRedemption } from './utils/serialize-redemption';
+
+@ApiTags('Redemption')
+@Controller('redemption')
+export class RedemptionController {
+  constructor(private readonly redemptionService: RedemptionService) {}
+
+  @ApiExcludeEndpoint()
+  @Post()
+  @UseGuards(MagicLinkGuard)
+  async create(
+    @Context() { user }: MagicLinkContext,
+    @Body(
+      new ValidationPipe({
+        errorHttpStatusCode: HttpStatus.UNPROCESSABLE_ENTITY,
+      }),
+    )
+    dto: RedemptionDto,
+  ): Promise<SerializedRedemption> {
+    if (await this.redemptionService.find(user)) {
+      throw new ForbiddenException('Redemption already exists');
```

Comment:
403 should be used if a user is authenticated but they don't have permissions for this operation. We're using 422 for user creation [here](https://github.com/iron-fish/ironfish-api/blob/master/src/users/users.service.ts#L131-L138). We should keep this consistent